### PR TITLE
Jiden/create generator sprotty 257

### LINKED ISSUE #257: Create generator-sprotty package for faster project creation
We could use [Yeoman](https://yeoman.io) to offer a project generator. This would help for creating new Sprotty projects more quickly and easily.

### PR DESCRIPTION
**Pull request by jonah-iden**
resolves #257
this implements a basic version of a yeoman generator for sprotty.
The example application this generator creats is based on the getting started example from https://github.com/gfontorbe/sprotty-website/pull/1

**How to  test**
1. `yarn`
2. `yarn build`
3. `npm link`
4. `yarn run run`

### CODE REVIEW

File: packages/generator-sprotty/sprotty-local-template/package.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    "description": "Please enter a brief description here",
+    "scripts": {
+      "build": "esbuild ./src/index.ts --bundle --sourcemap --outfile=./<%= out-path %>/index.js"
+    },
+    "devDependencies": {
+      "typescript": "4.9.5",
+      "esbuild": "^0.17.8"
+    },
+    "dependencies": {
+      "sprotty": "0.13.0",
```
Comment (spoenemann) [inline packages/generator-sprotty/sprotty-local-template/package.json:12]:
```suggestion
      "sprotty": "^0.13.0",
```

File: packages/generator-sprotty/sprotty-local-template/package.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
+    "name": "<%= project-name %>",
+    "description": "Please enter a brief description here",
+    "scripts": {
+      "build": "esbuild ./src/index.ts --bundle --sourcemap --outfile=./<%= out-path %>/index.js"
+    },
+    "devDependencies": {
+      "typescript": "4.9.5",
+      "esbuild": "^0.17.8"
```
Comment (spoenemann) [inline packages/generator-sprotty/sprotty-local-template/package.json:9]: Shall we add `http-server` and a start script?

Comment (jonah-iden) [reply packages/generator-sprotty/sprotty-local-template/package.json:9]: im not shure if we should. The example works with just opening the html file so i don't really see the need to add more dependencies that the user needs to remove later again when he wants to use the project for something else

File: packages/generator-sprotty/sprotty-local-template/src/views.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    render(node: Readonly<SNode & TaskNode>, context: RenderingContext): VNode {
+        return <g>
+            <rect class-sprotty-node={true} class-task={true}
+                class-running={node.isRunning}
+                class-finished={node.isFinished}
+                width={node.size.width}
+                height={node.size.height}
+            >
+            </rect>
+            <text x={50} y={50 + 5}>{node.name}</text>
```
Comment (gfontorbe) [inline packages/generator-sprotty/sprotty-local-template/src/views.tsx:19]:
```suggestion
            <text x={node.size.width / 2} y={node.size.height / 2 + 5}>{node.name}</text>
```

**Review by gfontorbe — COMMENTED** [review]
I don't think the .gitignore should be added to the package. At least not with its current content

**Comment by jonah-iden** [comment]
do you mean because of the "test-temp/"? 
I would definitetly leave app/ in there since thats the build output

File: packages/generator-sprotty/package.json [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    "files": [
+      "app",
+      "sprotty-local-template",
+      "src"
+    ],
+    "main": "app/index.js",
+    "types": "app/index.d.ts",
+    "scripts": {
+      "prepare": "yarn run clean && yarn run build",
+      "clean": "shx rm -rf app",
```
Comment (spoenemann) [inline packages/generator-sprotty/package.json:22]: We're currently using `rimraf` for that. It would be great to migrate to `shx`, but that should be done for all packages in a separate PR.

File: packages/generator-sprotty/sprotty-local-template/static/styles.css [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+.sprotty-node.task {
+    fill: #c0e0fc;
+    stroke: #444;
+    stroke-width: 1;
+}
+
+.sprotty-node.task.running {
+    fill: #f00;
```
Comment (spoenemann) [inline packages/generator-sprotty/sprotty-local-template/static/styles.css:13]: Could you change the example to use less aggressive colors?

File: packages/generator-sprotty/sprotty-local-template/static/index.html [hunk 6]
```diff
@@ -0,0 +1,7 @@
+<head>
+    <script src="index.js" type="text/javascript"></script>
+    <link rel="stylesheet" href="styles.css" />
+  </head>
+  <body>
+    <div id="<%= html-element-id %>"></div>
+  </body>
```
Comment (spoenemann) [inline packages/generator-sprotty/sprotty-local-template/static/index.html:7]: Please add `<!DOCTYPE html>` and proper indentation.

File: packages/generator-sprotty/src/index.ts [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+    }
+
+    async prompting(): Promise<void> {
+        this.answers = await this.prompt([
+            {
+                type: 'input',
+                name: 'projectName',
+                prefix: description(
+                    'Welcome to Sprotty!',
+                    'This tool generates a new Sprotty Project.\n',
```
Comment (spoenemann) [inline packages/generator-sprotty/src/index.ts:88]: The `description` function inserts spaces between the strings, so after my suggested change to add a line break here, the next line starts with a space. You could fix that by moving the next line into this one.

File: packages/generator-sprotty/src/index.ts [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+                    'The project name identifies the npm package and can be used by other packages to depend on this project.'
+                ),
+                message: 'Your project name:',
+                default: 'hello-world',
+            },
+            {
+                type: 'input',
+                name: 'mainElementId',
+                message: 'ID of your main html element for the sprotty diagram:',
```
Comment (spoenemann) [inline packages/generator-sprotty/src/index.ts:97]: Proposal: Shorten the message to 'Main HTML element id' and add a `prefix` with more explanations, e.g. 'Your Sprotty diagram will be rendered inside this element.'

File: packages/generator-sprotty/src/index.ts [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+                default: 'sprotty-diagram',
+            },
+            {
+                type: 'confirm',
+                name: 'generateStatic',
+                message: 'generate static folder with index.html and styles.css',
```
Comment (spoenemann) [inline packages/generator-sprotty/src/index.ts:103]: Proposal: Shorten the message to 'Generate static folder' and add a `prefix` with more explanations, e.g. 'Generate index.html and styles.css for a simple dev setup.'

**Review by spoenemann — APPROVED** [review]
Thank you!
